xapian-core/tests/perftest/perftest.cc

```diff
@@ -383,6 +384,47 @@ PerfTestLogger::searching_end()
     }
 }
 
+void
+PerfTestLogger::diversifying_start(const string & description)
+{
+    indexing_end();
+    searching_end();
+    write("   <diversifyrun>\n"
+	  "    <description>" + escape_xml(description) + "</description>\n");
+    diversifying_started = true;
+    diversify_start();
+}
+
+void
+PerfTestLogger::diversify_start()
+{
+    diversifying_timer = RealTime::now();
+}
+
+void
+PerfTestLogger::diversify_end(unsigned int k, const Xapian::DocumentSet & dset)
+{
+    Assert(diversifying_started);
+    double elapsed(RealTime::now() - diversifying_timer);
+    write("    <diversify>"
+	       "<time>" + str(elapsed) + "</time>"
+	       "<dset>"
+	       "<k>" + str(k) + "</k>"
+	       "<size>" + str(dset.size()) + "</size>"
+	       "</dset>"
+	       "</diversify>\n");
```

Comment:
The continuation lines here are overindented - they should line up vertically like:

```
    write("    <diversify>"
          "<time>" + str(elapsed) + "</time>"
          "<dset>"
```

Reply:
Modification pushed in a fixup commit.